### ISSUE TITLE
🐛 BUG: HRM not work with vue

### ISSUE THREAD

**Issue by diegoleme** [snowpackjs/astro#1843]
### What version of `astro` are you using?

0.21.0-next.3

### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

When I update the content of a .vue component, the component stops working and gives the following error:

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'querySelector')
    at onLoad (load.ts:8)
    at [product].astro?html-proxy&index=0.js:2
```

The error is caused because the `astro-root` tag `uid` is changed, but, in the `[product].astro?html-proxy&index=0.js` file the `uid` it doesn't.

The html example after change:
`<astro-root uid="2kIRFL">...</astro-root>`

The `[product].astro?html-proxy&index=0.js` after change:
```js
import setup from '/node_modules/.vite/astro_client_load_js.js?v=3b7eb40c';
setup("11qlWh", {value: true}, async () => {
    // ...
});

```

`2kIRFL` and `11qlWh` is not equal.

### Link to Minimal Reproducible Example

Não consigo reproduzir no Stackblitz and Sandbox

**Comment by natemoo-re** [comment]
Thanks for opening an issue! Going to close this as a duplicate of https://github.com/snowpackjs/astro/issues/1824, but this is very helpful context!

**Comment by natemoo-re** [comment]
After doing some digging, I narrowed down the issue in #1824 but this Vue-specific problem persists. Going to re-open this to track.

**Comment by diegoleme** [comment]
Bringing some more details.

This error only happens when the vue html template is changed.

**Comment by IanVS** [comment]
I'm experiencing something similar, but with React.  I'm not able to tell what's really going on, but I'll try to keep investigating.

**Comment by IanVS** [comment]
OK, I think I've been able to create a reproduction of the issue I'm having, sorry if it's a little messy.  I started with the docs theme, moved it to react, got it running on the `next.7` version, and can reproduce the issue.  

https://stackblitz.com/edit/github-adjrdl-lmdpgf?file=src/components/Header/LanguageSelect.jsx

You'll need to fork the stackblitz so you can make changes.  I was able to trigger the error by changing `Language` to `Lang` in the `LanguageSelect.jsx` file, then changing between the two languages.

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.

**Comment by diegoleme** [comment]
Yes, its work now! :)

Thank you
